src/main/perl/Lock.pm

```diff
@@ -107,16 +105,20 @@ sub unlock
     my $self = shift;
     if ($self->{LOCK_SET}) {
         # if we forced the lock LOCK_FH can be undef
+
         if ($self->{LOCK_FH}) {
             unless (flock($self->{LOCK_FH}, LOCK_UN)) {
                 $self->error("cannot release lock: $self->{LOCK_FILE}");
                 return;
             }
+            # close the filehandle, clearing any previous content
             $self->error("cannot close lock file: $self->{LOCK_FILE}")
                 unless $self->{LOCK_FH}->close();
```

Comment:
merging on one line would make it easier to read (or using an unless block): I had the feeling at the first look that the error was unconditionally raised....